models/chainlink/ethereum/chainlink_ethereum_ccip_onramps.sql

```diff
@@ -0,0 +1,93 @@
+{{
+    config(
+        tags=['dunesql'],
+        schema = 'chainlink_ethereum',
+        alias = alias('ccip_onramps'),
+        materialized = 'incremental',
+        file_format = 'delta',
+        unique_key = ['evt_index', 'evt_tx_hash', 'message'],
+        post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "chainlink",
+                                \'["synthquest"]\') }}'
+    )
+}}
+
+ {% set eth_router = '0xe561d5e02207fb5eb32cca20a699e0d8919a1476' %}
+ {% set test_date = "2022-06-01" %}
+
+select 
+'Ethereum Mainnet' as origin
+, 'Avalanche Mainnet' as destination
+, cast(json_extract_scalar(message, '$.sourceChainSelector') as INT256) as sourceChainSelector
+, cast(json_extract_scalar(message, '$.sequenceNumber') as INT256)  as sequenceNumber
+, cast(json_extract_scalar(message, '$.feeTokenAmount') as INT256) as feeTokenAmount
+, cast(json_extract_scalar(message, '$.sender') as VARCHAR) as sender
+, cast(json_extract_scalar(message, '$.nonce') as INT256) as nonce
+, cast(json_extract_scalar(message, '$.gasLimit') as INT256) as gasLimit
+, cast(json_extract_scalar(message, '$.strict') as BOOLEAN) as strict
+, cast(json_extract_scalar(message, '$.receiver') as VARCHAR) as receiver
+, cast(json_extract_scalar(message, '$.data') as varchar) as data
+, json_extract(message, '$.tokenAmounts') as tokenAmounts
```

Comment:
Not sure but we can try if this fix the test error:
```suggestion
, cast(json_extract_scalar(message, '$.tokenAmounts') as INT256) as tokenAmounts
```

Reply:
the `tokenAmounts` returns an array with json. I think there will be arrays that have >1 tokens sent with tx's in the future so this can't easily be parsed dynamically into columns either. It should stay as an array so we don't lose info or blow up the columns. Funny thing is it runs in dune. Maybe there's a dbt workaround to manage json columns

Example value:
["{\"token\":\"0x5947bb275c521040051d82396192181b413227a3\",\"amount\":1}"]

Reply:
Maybe we should try `array<bigint>` or something similar, but currently json data type may not be supported here.

Reply:
fingers crossed. I used array_agg & partitioned by a two columns that should create a unique ID for each tx. It should return as an array instead.